tests/fallback_test.py

```diff
@@ -0,0 +1,210 @@
+    "host_name",
+    "host.docker.internal",
+    "Host name all the services are bound on",
+)
+_NODE_ID = flags.DEFINE_string("node", "test-id", "Node ID")
+
+absl.flags.adopt_module_key_flags(framework.xds_k8s_testcase)
+
+
+def get_free_port() -> int:
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        sock.bind(("localhost", 0))
+        return sock.getsockname()[1]
+
+
+class FallbackTest(absltest.TestCase):
+    bootstrap: framework.helpers.docker.Bootstrap = None
+
+    @staticmethod
+    def setUpClass():
+        FallbackTest.bootstrap = framework.helpers.docker.Bootstrap(
+            framework.helpers.logs.log_dir_mkdir("bootstrap"),
+            ports=[get_free_port() for _ in range(2)],
+            host_name=_HOST_NAME.value,
+        )
+
+    def setUp(self):
+        logger.info("Starting %s", self.id())
+        self.process_manager = framework.helpers.docker.ProcessManager(
+            bootstrap=FallbackTest.bootstrap,
+            node_id=_NODE_ID.value,
+        )
+
+    def start_client(self, port: int = None, name: str = None):
+        logger.debug("Starting client process")
+        return framework.helpers.docker.Client(
+            manager=self.process_manager,
+            name=name or framework.xds_flags.CLIENT_NAME.value,
+            port=port or get_free_port(),
+            url="xds:///listener_0",
+            image=framework.xds_k8s_flags.CLIENT_IMAGE.value,
+        )
+
+    def start_control_plane(self, name: str, index: int, upstream_port: int):
+        logger.debug('Starting control plane "%s"', name)
+        return framework.helpers.docker.ControlPlane(
+            self.process_manager,
+            name=name,
+            port=self.bootstrap.xds_config_server_port(index),
+            upstream=f"{_HOST_NAME.value}:{upstream_port}",
+            image=_CONTROL_PLANE_IMAGE.value,
+        )
+
+    def start_server(self, name: str, port: int = None):
+        logger.debug('Starting server "%s"', name)
+        port = get_free_port() if port is None else port
+        return framework.helpers.docker.GrpcProcess(
+            name=name,
+            port=port,
+            image=framework.xds_k8s_flags.SERVER_IMAGE_CANONICAL.value,
+            manager=self.process_manager,
+            ports={8080: port},
+            command=[],
+        )
+
+    def test_fallback_on_startup(self):
+        with (
+            self.start_server(name="server1") as server1,
+            self.start_server(name="server2") as server2,
+            self.start_client() as client,
+        ):
+            self.assertTrue(
+                client.expect_message_in_output(
+                    "UNAVAILABLE: xDS channel for server"
+                )
+            )
+            self.assertEqual(client.get_stats(5).num_failures, 5)
+            # Fallback control plane start, send traffic to server2
+            with self.start_control_plane(
+                "fallback_xds_config", 1, server2.port
```

Comment:
Can we add key to the params like we did for `primary_xds_config`?

Reply:
Done.